lib/constructor.js

```diff
@@ -109,9 +109,19 @@ function size_of(segments, bindings) {
   return size;
 }
 
+function writeBinaryString(inputString) {
+  return (inputString.match(/.{2}/g) || [])
+    .map((value) => {
+      return parseInt(value, 16);
+    });
+}
+
 function writeBinary(segment, buf, offset, bindings) {
   var bin = bindings[segment.name];
   var size = size_of_segment(segment, bindings);
+  if (typeof bin === 'string') {
```

Comment:
please try to use format: "string-hexbin", which is supposed to do what you seem to want